x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,784 @@
+start a <<restart-cluster-rolling,rolling restart>>.
++
+Stop at the step indicating *Perform any needed changes*, and then proceed to
+the next step in this procedure.
+
+. Replace your existing keystore with the new keystore, ensuring that the
+file names match. For example, `node1-http.p12`.
++
+If your 
+<<cert-password-updates,keystore password is changing>>, then save the
+keystore with a new filename so that {es} doesn't attempt to reload the file
+before you update the password.
+
+. If you needed to save the new keystore with a new filename, update the
+`ES_PATH_CONF/elasticsearch.yml` file to use the filename of the new keystore.
+For example:
++
+[source,yaml]
+----
+xpack.security.http.ssl.enabled: true
+xpack.security.http.ssl.keystore.path: node1-http.p12
+----
+
+. If your keystore password is changing, add the password for your private key
+to the secure settings in {es}.
++
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.http.ssl.keystore.secure_password
+----
+
+. Start the node where you updated the keystore, and then complete the remaining
+steps for a <<restart-cluster-rolling,rolling restart>>.
+
+. [[verify-keystore-http,step 10]](Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that
+{es} loaded the new keystore.
+
+. One node at a time, complete <<start-rolling-restart-http>> through
+<<verify-keystore-http>> until you've updated all keystores in your cluster.
+
+[[update-node-certs-different]]
+=== Update security certificates with a different CA
+++++
+<titleabbrev>With a different CA</titleabbrev>
+++++
+If you have to trust a new CA, generate a new CA certificate and key for each
+node that is signed by the new CA certificate and instruct your nodes to trust
+this certificate chain.
+
+[[node-certs-different-transport]]
+==== Generate a new certificate for the transport layer
+Create a new CA certificate and add it to your existing CA truststore. After
+{es} reloads the file, you can remove the old CA certificate from your
+truststore.
+
+NOTE: The following examples use PKCS#12 files, but the same steps apply to JKS
+keystores.
+
+. Open the `ES_PATH_CONF/elasticsearch.yml` file and check the names of the
+security certificates that are currently in use. You'll use the same names
+for your new certificates.
++
+In this example, the keystore and truststore are using different files.
+Your configuration might use the same file for both the keystore and the 
+truststore. Elastic recommends using separate files. 
++
+[source,yaml]
+----
+xpack.security.transport.ssl.keystore.path: config/elastic-certificates.p12
+xpack.security.transport.ssl.keystore.type: PKCS12
+xpack.security.transport.ssl.truststore.path: config/elastic-stack-ca.p12
+xpack.security.transport.ssl.truststore.type: PKCS12
+----
+
+. On *any* node in your cluster, generate a new CA certificate. You only need
+to complete this step one time.
++
+NOTE: If your organization has its own CA, you'll need to 
```

Comment:
Lets put that note _after_ the CA generation. This is related to nodes certifcate generation , not the CA certificate generation, so it might confuse users that we mention it here